backends/gdx-backend-lwjgl3/src/com/badlogic/gdx/backends/lwjgl3/Lwjgl3ApplicationConfiguration.java

```diff
@@ -337,4 +350,54 @@ static Lwjgl3Monitor toLwjgl3Monitor(long glfwMonitor) {
 		String name = GLFW.glfwGetMonitorName(glfwMonitor);
 		return new Lwjgl3Monitor(glfwMonitor, virtualX, virtualY, name);
 	}
+
+	/**
+	 * Returns where preferences are stored by default.
+	 * Typically AppData/Roaming on Windows, Library/Preferences on macOS and .config on Linux.
+	 * @return The default preferences directory.
+	 */
+	public String getDefaultPreferencesDirectory() {
+
+		if (UIUtils.isWindows) {
+			String appdata = System.getenv("APPDATA");
+			String windir = System.getenv("WINDIR");
+			return (appdata != null) ? appdata // 2000/XP/Vista/7/8/10/11
+				: (windir != null) ? windir + "/Application Data" // 95/98/Me
+				: preferencesBackupDirectory; // Default to backup directory (probably ~/.prefs/) if it's broken
+
+		} else if (UIUtils.isMac) {
+			return "Library/Preferences";
+
+		} else if (UIUtils.isLinux) {
+			String configHome = System.getenv("XDG_CONFIG_HOME");
+			if (configHome != null) {
+				//TODO: Someone with regex skills should make this work with all environment variables
+				if (configHome.contains("$HOME")) configHome = configHome.replace("$HOME", System.getenv("HOME"));
+				else if (configHome.contains("$")) configHome = null;

```

Comment:
I can try some things for the regex. I don't like forcing `configHome` to null if any non-HOME environment variables are present, or even any escaped dollar signs.

Reply:
This is pretty gnarly, but I think it works, based on limited testing in jshell. It should ignore backslash-escaped dollar signs, treating them as literal dollar signs.

```java
Pattern p = Pattern.compile("(?<!\\\\)\\$(\\w+)");
Matcher m = p.matcher(configHome);
while(m.find()) {
  m.reset(configHome = configHome.replaceFirst("\\Q" + m.group() + "\\E",
    Matcher.quoteReplacement(String.valueOf(System.getenv(m.group(1))))));
}
```

Reply:
The above code can replace the highlighted range in Lwjgl3ApplicationConfiguration, and the similar range with a TODO in
LwjglApplicationConfiguration. That's the `if` and its `else if`.

Reply:
Sorry I didn't notice this for a while. I've committed this now. Nighty night.

Reply:
Apparently typing `$XDG_CONFIG_HOME` in my terminal wasn't a good idea. The commit messages are screwed.

Reply:
Great! Consider this resolved.